testing/integration/serverless/serverless_test.go

```diff
@@ -0,0 +1,46 @@
+// Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+// or more contributor license agreements. Licensed under the Elastic License 2.0;
+// you may not use this file except in compliance with the Elastic License 2.0.
+
+package serverless
+
+import (
+	"flag"
+	"log"
+	"os"
+	"testing"
+
+	"github.com/elastic/elastic-agent/pkg/testing/define"
+)
+
+var flagSet = flag.CommandLine
+
+func init() {
+	define.RegisterFlags("integration.", flagSet)
+}
+
+func TestMain(m *testing.M) {
+	flag.Parse()
+
+	if define.AutoDiscover {
+		define.InitAutodiscovery(nil)
+	}
+
+	runExitCode := m.Run()
+
+	if define.AutoDiscover {
+		discoveredTests, err := define.DumpAutodiscoveryYAML()
+		if err != nil {
+			log.Println("Error dumping autodiscovery YAML:", err)
+			os.Exit(1)
+		}
+
+		err = os.WriteFile(define.AutoDiscoveryOutput, discoveredTests, 0644)
+		if err != nil {
+			log.Printf("Error writing autodiscovery data in %q: %s", define.AutoDiscoveryOutput, err)
+			os.Exit(1)
+		}
+	}
+
+	os.Exit(runExitCode)
```

Comment:
At the moment, the implementation `TestMain` is identical between `leak` and `serverless` packages.  If possible, it would be good to share the implementation code to prevent unintentional divergence (e.g. https://github.com/elastic/elastic-agent/pull/8396/files#r2136172728 and https://github.com/elastic/elastic-agent/pull/8396/files#r2136171866).

Reply:
IMO we want to allow divergence, as the k8s integration tests will have to diverge in the `TestMain` 🙂 How about we apply DRY, when all remaining integration tests are split and we can make a clearer decision then, wdyt?